onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -276,6 +278,7 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
     string string_buf;
     model_proto.SerializeToString(&string_buf);
 
+    std::lock_guard<OrtMutex> glock(tensorrt_mu);
```

Comment:
why is a mutex needed here?

Reply:
Removed the mutex in compile and add one just before enqueue in compute